bn-apps/cordapp-updates-distribution/corda-updates-app/src/main/kotlin/net/corda/cordaupdates/app/member/SyncArtifactsFlow.kt

```diff
@@ -0,0 +1,124 @@
+ * @param launchAsync flag that indicates whether the synchronisation should be started asynchronously.
+ *      Synchronous invocations shouldn't be used for Corda-based transports.
+ *
+ */
+@SchedulableFlow
+@StartableByRPC
+class SyncArtifactsFlow private constructor (private val scheduledStateRef : StateRef? = null,
+                         private val syncerConfig : SyncerConfiguration? = null,
+                         private val launchAsync : Boolean = true) : FlowLogic<List<ArtifactMetadata>?>() {
+    // this constructor is needed when the flow is invoked by Corda scheduler
+    constructor(scheduledStateRef : StateRef) : this(scheduledStateRef, null, true)
+    constructor(syncerConfig : SyncerConfiguration? = null,
+                launchAsync : Boolean = true) : this(null, syncerConfig, launchAsync)
+
+    // TODO: remove the progress tracker once Corda v4 is released
+    companion object {
+        object LAUNCHING_SYNCHRONISATION : ProgressTracker.Step("Invoking initial synchronisation")
+
+        fun tracker() = ProgressTracker(
+                LAUNCHING_SYNCHRONISATION
+        )
+    }
+
+    override val progressTracker = tracker()
+
+    @Suspendable
+    override fun call() : List<ArtifactMetadata>? {
+        val syncerService = serviceHub.cordaService(SyncerService::class.java)
+        progressTracker.currentStep = LAUNCHING_SYNCHRONISATION
+        return if (launchAsync) {
+            syncerService.syncArtifactsAsync(syncerConfig)
+            null
+        } else {
+            syncerService.syncArtifacts(syncerConfig)
+        }
+    }
+}
+
+/**
+ * Issues a [ScheduledSyncState] onto the ledger (all existing [ScheduledSyncState]s will be spent first) and then triggers
+ * a synchronisation via [SyncArtifactsFlow]
+ */
+@StartableByRPC
+class ScheduleSyncFlow @JvmOverloads constructor(private val syncerConfig : SyncerConfiguration? = null,
+                                                 private val launchAsync : Boolean = true) : FlowLogic<List<ArtifactMetadata>?>() {
+
+    // TODO: remove the progress tracker once Corda v4 is released
+    companion object {
+        object SCHEDULING_STATE : ProgressTracker.Step("Scheduling state")
+        object INVOKING_INITIAL_SYNCRONISATION : ProgressTracker.Step("Invoking initial synchronisation") {
+            override fun childProgressTracker() = SyncArtifactsFlow.tracker()
+        }
+
+        fun tracker() = ProgressTracker(
+                SCHEDULING_STATE,
+                INVOKING_INITIAL_SYNCRONISATION
+        )
+    }
+
+    override val progressTracker = tracker()
+
+    @Suspendable
+    override fun call() : List<ArtifactMetadata>? {
+        progressTracker.currentStep = SCHEDULING_STATE
+        val configuration = serviceHub.cordaService(MemberConfiguration::class.java)
+
+        val scheduledStates = serviceHub.vaultService.queryBy<ScheduledSyncState>().states
+
+        // spending all existing scheduled states
```

Comment:
Maybe clarify here that we're not really "spending" or "issuing" anything here, just deleting the old scheduled task and creating a new one. E.g. with a comment or better naming.

Reply:
Done. Also renamed `spendScheduledState` to `removeScheduledState` for more clarity